### ISSUE TITLE
phpdocumentor fatal error since drush 8.1.5

### ISSUE THREAD

**Issue by accuraz** [drush-ops/drush#2365]
I'll get the same fatal error on every drush command since drush 8.1.5
</br>
That's the log :
</br>
```
$ drush cron
PHP Fatal error:  phpDocumentor\Reflection\DocBlock\Tags\BaseTag cannot implement phpDocumentor\Reflection\DocBlock\Tag - it is not an interface in /var/www/exemple/vendor/phpdocumentor/reflection-docblock/src/DocBlock/Tags/BaseTag.php on line 21
Drush command terminated abnormally due to an unrecoverable error.   [error]
Error: phpDocumentor\Reflection\DocBlock\Tags\BaseTag cannot
implement phpDocumentor\Reflection\DocBlock\Tag - it is not an
interface in
/var/www/exemple/vendor/phpdocumentor/reflection-docblock/src/DocBlock/Tags/BaseTag.php,
line 21
```

Is this some king of a bug or maybe can someone help me out ?
</br>
Thanks,

**Comment by weitzman** [comment]
Are you using a global Drush and/or a Drush local to your project via
composer.json? How did you upgrade? Make sure you do a composer update to
get new dependencies.

**Comment by accuraz** [comment]
I'm using a global drush and I ran composer update after I get the error message and before I post this issue.
</br>
If it might help, I run php7

**Comment by weitzman** [comment]
What version of Drupal? Can you run `drush status --debug` when you are not in a drupal directory?

**Comment by accuraz** [comment]
I run Drupal 8.1.10
</br>
I can run `drush status --debug` when I'm not into a Drupal directory with no errors at output
</br>
If I run `drush status --debug` into a Drupal directory I'm getting the same errors about phpdocumentor

**Comment by sms20** [comment]
Yeah got the same issue as well.

Steps to replicate:
1) Install Drupal 8.1.0 
2) Install AMP Module (https://www.drupal.org/project/amp)
3) Run composer drupal-update
4) Run drush status - now reports error as above

Running PHP 5.5.9

**Comment by greg-1-anderson** [comment]
Drush uses phpDocumentor, but it can use either version 2 or version 3. The trouble is, if you are using a global Drush, it will be using version 3.  This will cause problems for you if your Drupal site is using a module that uses version 2.

Two solutions:

1. Install global Drush via git clone + composer install, then use `composer require` to specifically request version 2 of phpDocumentor.

2. Install a site-local Drush with your Drupal site.  Composer should then automatically install version 2 of phpDocumentor -- presuming that you installed Drupal + the amp module via drupal-composer/drupal-project.

Didn't try to reproduce any of this -- that's just what it looks like to me, based on the symptoms.

**Comment by sms20** [comment]
Installing Drush via composer worked for me as documented here:

**Install a global Drush via Composer** - http://www.drush.org/en/master/install-alternative/

Thanks.

**Comment by JTubex** [comment]
Drupal itself uses version 2.0.4 of phpDocumentor: http://cgit.drupalcode.org/drupal/tree/composer.lock

So I think we should fix this in Drush.

**Comment by accuraz** [comment]
Ok guys the composer installation method works for me to but on the other and @JTubex is right about the fact that Drupal 8.1.x uses phpDocumrntor 2.0.5
</br>
So it seems to look like a Drush related problem...

**Comment by greg-1-anderson** [comment]
Looks like we should add a constraint for phpdocumentor ~2 in Drush.

**Comment by weitzman** [comment]
Fixed by #2369
